File: app/api/v1/endpoints/monitoring.py
# app/api/v1/endpoints/monitoring.py
import sys

sys.path.insert(0, "/home/bamer/llama-runner-async-proxy")

from fastapi import APIRouter
from typing import Dict, Any

# Import after path setup
from llama_runner_legacy.services.metrics_collector import MetricsCollector

router = APIRouter()

# Singleton collector
from llama_runner_legacy.services.metrics_collector import GLOBAL_METRICS_COLLECTOR

collector = GLOBAL_METRICS_COLLECTOR or MetricsCollector()


@router.get("/summary", summary="Get real‑time metrics summary")
async def get_summary() -> Dict[str, Any]:
    """Return current metrics snapshot."""
    # Get the summary from the collector
    summary = collector.get_summary()

    # Ensure we return a proper object structure even if null
    if not summary:
        return {
            "uptime": 0,
            "total_models": 0,
            "active_runners": 0,
            "total_starts": 0,
            "total_stops": 0,
            "total_errors": 0,
            "memory_usage": {"current": "N/A", "peak": "N/A"},
            "load_average": "N/A",
        }

    return summary

File: backend/src/routes/monitoring.js
const express = require('express');
const router = express.Router();

// Mock metrics collector - in real implementation this would be a proper metrics system
let metricsCollector = {
    get_summary: () => ({
        uptime: 120,
        total_models: 5,
        active_runners: 3,
        total_starts: 10,
        total_stops: 2,
        total_errors: 0,
        memory_usage: { current: "1.2GB", peak: "2.4GB" },
        load_average: "0.75"
    })
};

// GET /api/v1/monitoring/summary
router.get('/monitoring/summary', (req, res) => {
    const summary = metricsCollector.get_summary();
    res.json(summary);
});

module.exports = router;